### PR TITLE
[#12290][fix] Qwen 3.5 fix 3d position ID handling

### LINKED ISSUE #12271: [Bug]: qwen3_5_moe model type missing multimodal placeholder registration causing "Unknown modality: image" error
### System Info

  - TensorRT-LLM: commit 390a7fd6b
  - Model: Qwen3.5-35B-A3B
  - GPU: RTX 4090 x 2
  - Backend: _autodeploy

### Who can help?

_No response_

### Information

- [ ] The official example scripts
- [ ] My own modified scripts

### Tasks

- [ ] An officially supported task in the `examples` folder (such as GLUE/SQuAD, ...)
- [ ] My own task or dataset (give details below)

### Reproduction

Problems:
When serving Qwen3.5 MoE vision-language models (e.g., Qwen3.5-35B-A3B) using trtllm-serve with _autodeploy backend, multimodal image/video requests fail with:  Exception: Error code: 400 - {'object': 'error', 'message': 'Unknown modality: image', 'type': 'BadRequestError', 'param': None, 'code': 400} 

Root Cause:
The qwen3_5_moe model type is not registered in MULTIMODAL_PLACEHOLDER_REGISTRY, so add_multimodal_placeholders() in tensorrt_llm/inputs/utils.py throws "Unknown modality" error.                                                                                                                                                                                    
  Although Qwen3_5MoeForConditionalGeneration in tensorrt_llm/_torch/auto_deploy/models/custom/modeling_qwen3_5_moe.py implements full vision-language capabilities (pixel_values,
  image_grid_thw, etc.), it model type lacks multimodal placeholder registration required by the serving layer.

### Expected behavior

Add @register_input_processor decorator to Qwen3_5MoeForConditionalGeneration class to register image/video placeholders:
  @register_input_processor(
      BaseMultimodalInputProcessor,
      model_type="qwen3_5_moe",
      placeholder_metadata=MultimodalPlaceholderMetadata(
          placeholder_map={
              "image": "<|vision_start|><|image_pad|><|vision_end|>",
              "video": "<|vision_start|><|video_pad|><|vision_end|>",
          },
          placeholder_placement=MultimodalPlaceholderPlacement.BEFORE_TEXT,
          placeholders_separator="",
      ),
  )
  class Qwen3_5MoeForConditionalGeneration(Qwen3_5MoePreTrainedModel):
      # ... existing code ...

### actual behavior

None of that was added now.

### additional notes

Nothing to add 

### Before submitting a new issue...

- [x] Make sure you already searched for relevant issues, and checked the [documentation](https://nvidia.github.io/TensorRT-LLM/) and [examples](https://github.com/NVIDIA/TensorRT-LLM/tree/main/examples) for answers to frequently asked questions.

### PR DESCRIPTION
**Pull request by bmarimuthu-nv**
## Description

Fixes https://github.com/NVIDIA/TensorRT-LLM/issues/12271 

This updates the Qwen3.5 MoE AutoDeploy multimodal path to make mRoPE delta handling request-scoped and cache-backed, while keeping Qwen-specific multimodal span construction in the Qwen model path instead of the generic input processor.

Implemented changes:
- derive exact Qwen multimodal spans from the tokenized `input_ids` in the Qwen-specific AutoDeploy path
- build chunk-aware multimodal 3D positions in the VLM wrapper
- add request-level mRoPE delta helpers and wire the wrapper to use a per-request `mrope_delta_cache` when available
- add the `initialize_qwen_mrope_delta_cache` transform to allocate a tiny per-slot cache resource
- make metadata-free prefill warmups a zero-delta no-op for the cached op
- fix the flattened cached `position_ids` fallback so request-level deltas are expanded to per-token deltas using `cu_seqlen`
- remove the temporary reduced-layer cap in the Qwen3.5 35B MoE AutoDeploy accuracy test after the e2e path was revalidated

## Test Coverage

Verified locally:
- `pytest -q tests/unittest/_torch/auto_deploy/unit/singlegpu/models/test_qwen3_5_moe.py -k "mrope_delta_matches_compute_mrope_positions_for_mixed_items or mrope_delta_with_cache_writes_prefill_and_reads_decode or flattened_prefill_with_request_deltas" -p no:cacheprovider`
- `TRTLLM_ACCURACY_NO_REFERENCE=1 TMPDIR=/tmp/trtqwen HF_HOME=/tmp/trtqwen/hf_home PYTHONDONTWRITEBYTECODE=1 pytest -q tests/integration/defs/accuracy/test_llm_api_autodeploy.py::TestQwen3_5_35B_MoE::test_bf16 -s`

Result from the full-model MMMU run:
- `mmmu_val` average accuracy: `40.11`
- test status: `1 passed`

## PR Checklist

Please review the following before submitting your PR:
- PR description clearly explains what and why. If using CodeRabbit's summary, please make sure it makes sense.
- PR Follows [TRT-LLM CODING GUIDELINES](https://github.com/NVIDIA/TensorRT-LLM/blob/main/CODING_GUIDELINES.md) to the best of your knowledge.
- Test cases are provided for new code paths (see [test instructions](https://github.com/NVIDIA/TensorRT-LLM/tree/main/tests#1-how-does-the-ci-work))
- Any new dependencies have been scanned for license and vulnerabilities
- [CODEOWNERS](https://github.com/NVIDIA/TensorRT-LLM/blob/main/.github/CODEOWNERS) updated if ownership changes
- Documentation updated as needed
- Update [tava architecture diagram](https://github.com/NVIDIA/TensorRT-LLM/blob/main/.github/tava_architecture_diagram.md) if there is a significant design change in PR.
- The reviewers assigned automatically/manually are appropriate for the PR.

- [x] Please check this after reviewing the above items as appropriate for this PR.

## GitHub Bot Help

To see a list of available CI bot commands, please comment `/bot help`.

## Release Notes

* **New Features**
  * Enhanced Qwen 3.5 MoE model support with multimodal position handling capabilities.
  * Added automatic model factory registration for improved deployment configuration.

* **Tests**
  * Comprehensive multimodal model test coverage including position computation and caching scenarios.
  * New validation tests for deployment transform initialization and sharding configuration.

### CODE REVIEW

**Comment by bmarimuthu-nv** [comment]
/bot run --extra-stage "DGX_B200-4_GPUs-AutoDeploy-1, DGX_H100-4_GPUs-AutoDeploy-1" --disable-fail-fast

**Comment by bmarimuthu-nv** [comment]
/bot run --extra-stage "DGX_B200-4_GPUs-AutoDeploy-1, DGX_H100-4_GPUs-AutoDeploy-1" --disable-fail-fast --reuse-test

File: examples/auto_deploy/model_registry/configs/qwen3.5_moe_400b.yaml [hunk 1]
```diff
@@ -1,31 +1,38 @@
...
   detect_sharding:
-    allreduce_strategy: SYMM_MEM
-  multi_stream_moe:
-    stage: compile
-    enabled: true
-  multi_stream_gemm:
-    stage: compile
-    enabled: true
-  gather_logits_before_lm_head:
-    enabled: true
```
Comment (suyoggupta) [inline examples/auto_deploy/model_registry/configs/qwen3.5_moe_400b.yaml]: ditto.. these are important for perf.

File: examples/auto_deploy/model_registry/configs/qwen3.5_moe_35b.yaml [hunk 2]
```diff
@@ -1,31 +1,35 @@
...
   detect_sharding:
-    allreduce_strategy: SYMM_MEM
-  multi_stream_moe:
-    stage: compile
-    enabled: true
-  multi_stream_gemm:
-    stage: compile
-    enabled: true
-  gather_logits_before_lm_head:
-    enabled: true
```
[the remark this answers is not in the document]
Comment (bmarimuthu-nv) [reply examples/auto_deploy/model_registry/configs/qwen3.5_moe_35b.yaml]: this should have been there, incorrectly removed while rebasing and consolidating the config.yaml. Fixing it

**Review by lucaslie — APPROVED** [review]

**Review by xinhe-nv — APPROVED** [review]

**Comment by bmarimuthu-nv** [comment]
/bot run --extra-stage "DGX_B200-4_GPUs-AutoDeploy-1, DGX_H100-4_GPUs-AutoDeploy-1" --disable-fail-fast --reuse-test

**Comment by bmarimuthu-nv** [comment]
/bot run --extra-stage "DGX_B200-4_GPUs-AutoDeploy-1, DGX_H100-4_GPUs-AutoDeploy-1" --disable-fail-fast

**Comment by bmarimuthu-nv** [comment]
/bot run --disable-fail-fast --extra-stage "DGX_B200-4_GPUs-AutoDeploy-1, DGX_H100-4_GPUs-AutoDeploy-1"

**Review by nvchenghaoz — APPROVED** [review]
